src/components2/MapLegend/LegendStyles.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import config from '~/config';
+import { media } from '~/styles/utils';
+
+export const LegendWrapper = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  flex-direction: column;
+
+  ${media.s`
+    flex-direction: row;
+  `}
+`;
+
+export const Header = styled.h2`
+  font-size: 0.9rem;
+  flex-basis: 100%;
+`;
+
+export const HeaderCol = styled(Header)`
+  flex-basis: max-content;
+`;
+
+export const LegendCol = styled.div`
+  align-items: flex-start;
+  display: flex;
+  flex: 1;
+  flex-direction: column;
+  margin-right: 10px;
+`;
+
+export const LegendItem = styled.div`
+  align-items: center;
+  color: ${config.colors.darkbg};
+  display: flex;
+  flex-direction: row;
+  font-size: 0.9em;
+  line-height: 1.16;
+  height: 2.2rem;
+
+  & img {
+    margin-right: 10px;
+  }
+`;
+
+export const CadastreLegendItem = styled.span`
+  min-width: 27px;
+  height: 18px;
+  margin: 5px 10px 5px 0;
+
+  ${media.s`
+    margin: 7px 20px 6px 0;
+  `}
+`;
+
+export const CadastreLegendItemColor = styled(CadastreLegendItem)<{
+  legendColor: string;
+}>`
+  background: ${(props) => props.legendColor};
+`;
+
+const CadastreLegendItemPointInner = styled.div<{
+  borderColor: string;
+  innerColor: string;
+}>`
+  border: solid 2px ${(props) => props.borderColor};
+  background-color: ${(props) => props.innerColor};
+  width: 8px;
+  height: 8px;
+  border-radius: 50%;
+  margin: auto;
+`;
+
+export const CadastreLegendItemPoint = (props) => (
+  <CadastreLegendItem>
+    <CadastreLegendItemPointInner {...props} />
+  </CadastreLegendItem>
+);
```

Comment:
Siehe oben, besser ist es explizit zu übergeben.

```suggestion
type = CadastreLegendItemPointInner = {
  borderColor: string;
  innerColor: string;
}

const CadastreLegendItemPointInner = styled.div<CadastreLegendItemPointInner>`
  border: solid 2px ${(props) => props.borderColor};
  background-color: ${(props) => props.innerColor};
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin: auto;
`;

export const CadastreLegendItemPoint: React.FC<CadastreLegendItemPointInner> = ( { borderColor, innerColor } ) => (
  <CadastreLegendItem>
    <CadastreLegendItemPointInner borderColor={ borderColor } innerColor={ innerColor } />
  </CadastreLegendItem>
);
```

Reply:
Looks like `React.FC` is not the preferable way to go: https://stackoverflow.com/questions/59988667/typescript-react-fcprops-confusion

Reply:
Interesting, thanks!

Some reading I found helpful:
- https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/
- https://www.harrymt.com/blog/2020/05/20/react-typescript-react-fc.html
- https://www.mydatahack.com/using-react-vfc-instead-of-react-fc/

My take:
- We should stick with the pattern `const FunctionName: React.VFC<Props> = ({…}) => {…}` whenever possible
- But use `React.VFC` whenever we don't use `children` (or want to type them explicitly).
- I find this easier to read than typing inside the brackes `({…}: {…})` or `({…}: Props)` and it gives additional React-specific stuff out of the box.
- We don't use defaultProps, so that's not an issue
- Having the `displayName` handled automatically is good for our StyledComponents-Based React
- Handling the `Sub.Component` pattern is also possible, so IMO not an issue.

I would not update the Codebase, though, but change it on the go, since the main issue is being explicit about `children` which is not that big of deal.